src/test/java/test/sanitycheck/CheckSuiteNamesTest.java

```diff
@@ -81,4 +88,26 @@ public void checkXmlSuiteAddition() throws ParserConfigurationException, SAXExce
     tng.setXmlSuites(parser.parseToList());
     tng.initializeSuitesAndJarFile();		
   }
+  
+  @Test(description = "Verify that same suite is not added multiple times")
+  public void validateDuplicateSuiteAddition() throws ParserConfigurationException, SAXException, IOException
```

Comment:
Same child suite will point to same tests, same tests will fail the check in checkTestNames this method. That is the confusion I have. I am seeing the exception about same test name for two tests in one suite. Lets deal with this in a seperate issue. I will open one for this.